libcudacxx/include/cuda/__annotated_ptr/access_property.h

```diff
@@ -0,0 +1,287 @@
+ * NVIDIA, 2788 San Tomas Expressway, Santa Clara, CA 95051.
+ *
+ * 15. ENTIRE AGREEMENT. This license is the final, complete and exclusive agreement between the parties relating to the
+ * subject matter of this license and supersedes all prior or contemporaneous understandings and agreements relating to
+ * this subject matter, whether oral or written. If any court of competent jurisdiction determines that any provision of
+ * this license is illegal, invalid or unenforceable, the remaining provisions will remain in full force and effect.
+ * This license may only be modified in a writing signed by an authorized representative of each party.
+ *
+ * (v. August 20, 2021)
+ */
+
+#ifndef _CUDA___ANNOTATED_PTR_ACCESS_PROPERTY
+#define _CUDA___ANNOTATED_PTR_ACCESS_PROPERTY
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda_runtime_api.h>
+
+#include <cuda/__annotated_ptr/access_property_encoding.h>
+#include <cuda/std/cstddef>
+#include <cuda/std/cstdint>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+template <typename>
+class __annotated_ptr_base; // forward declaration
+
+class access_property
+{
+private:
+  uint64_t __descriptor = __l2_interleave_normal;
+
+  friend class __annotated_ptr_base<access_property>;
+
+  // needed by __annotated_ptr_base
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr access_property(uint64_t __descriptor1) noexcept
+      : __descriptor{__descriptor1}
+  {}
+
+public:
+  struct shared
+  {};
+  struct global
+  {};
+  struct persisting
+  {
+    [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr operator cudaAccessProperty() const noexcept
+    {
+      return cudaAccessProperty::cudaAccessPropertyPersisting;
+    }
+  };
+  struct streaming
+  {
+    [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr operator cudaAccessProperty() const noexcept
+    {
+      return cudaAccessProperty::cudaAccessPropertyStreaming;
+    }
+  };
+  struct normal
+  {
+    [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr operator cudaAccessProperty() const noexcept
+    {
+      return cudaAccessProperty::cudaAccessPropertyNormal;
+    }
+  };
+
+  _CCCL_HIDE_FROM_ABI access_property() noexcept                                  = default;
+  _CCCL_HIDE_FROM_ABI access_property(const access_property&) noexcept            = default;
+  _CCCL_HIDE_FROM_ABI access_property(access_property&&) noexcept                 = default;
+  _CCCL_HIDE_FROM_ABI access_property& operator=(const access_property&) noexcept = default;
+  _CCCL_HIDE_FROM_ABI access_property& operator=(access_property&&) noexcept      = default;
+  _CCCL_HIDE_FROM_ABI ~access_property() noexcept                                 = default;
```

Comment:
We should only need one
```suggestion
  _CCCL_HIDE_FROM_ABI access_property() noexcept                                  = default;
```

Reply:
ok, clarified this point. https://en.cppreference.com/w/cpp/language/noexcept_spec
All members/base classes are no throwing, so the class doesn't need all the explicit definitions.